ewmh.c

```diff
@@ -216,6 +216,7 @@ ewmh_init(void)
     luaA_class_connect_signal(L, &client_class, "property::fullscreen" , ewmh_client_update_hints);
     luaA_class_connect_signal(L, &client_class, "property::maximized_horizontal" , ewmh_client_update_hints);
     luaA_class_connect_signal(L, &client_class, "property::maximized_vertical" , ewmh_client_update_hints);
+    luaA_class_connect_signal(L, &client_class, "property::maximized" , ewmh_client_update_hints);
```

Comment:
The previous version of the patch kept the 2bit==maximized concept, but it doesn't work after all.

*problem 1:*
When you maximized something, all 3 signals were sent. If some code handled "maximized" a "maximized_horizontally" (or all 3), then it was executed 3 times.

*problem 1.1:*
`request::geometry` was sent twice (before this PR) when you did a single operation. This is why it failed to serialize the size.

*problem 2:*
If you had the 3 maximize button in the titlebar, than where you toggled maximize twice, it would override the other 2 and do->undo would not restore the state as it was.

So, having a split state made most operation *more* completed than having 3 independent states. Also, have 3 booleans ensure that when changing one of them, only 1 signal is emitted. The previous code worked, I am not denying that, but it had the "design quirks build-in" while the request API blueyed and me implemented is designed to unify "concepts" into a single handler. It *could* be fixed in the handler, but I would rather have a simplified pipeline. This is what this patch does. Most users use maximize and not the other 2, so simplifying that one while making the other 2 a little more complex doesn't seem such a bad idea.

> How about deprecating the horizontal/vertical maximization instead and only have a single "maximized" state instead?

I don't think its necessary. Once this patch becomes stable, it wont need to be changed. I probably should merge `fullscreen` into this function as well. I will if I ever attack #758 and [insert z-index limitations issue I can't find right now].